apps/main/src/loan/components/PageCrvUsdStaking/Statistics/RevenueChartFooter.tsx

```diff
@@ -0,0 +1,68 @@
+import type { TimeOption } from '@ui-kit/lib/types/scrvusd'
+import Stack from '@mui/material/Stack'
+import ToggleButtonGroup from '@mui/material/ToggleButtonGroup'
+import ToggleButton from '@mui/material/ToggleButton'
+import Typography from '@mui/material/Typography'
+import LegendLine from '@/loan/components/PageCrvUsdStaking/Statistics/components/LegendLine'
+import { priceLineLabels } from '@/loan/components/PageCrvUsdStaking/Statistics/constants'
+import { useTheme } from '@mui/material/styles'
+import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
+
+const { Spacing } = SizesAndSpaces
+
+type Props = {
+  timeOptions: TimeOption[]
+  activeTimeOption: TimeOption
+  setActiveTimeOption: (event: React.MouseEvent<HTMLElement>, newTimeOption: TimeOption) => void
+}
+
+const LegendSet = ({ label, dash, legendColor }: { label: string; dash: string; legendColor: string }) => (
+  <Stack direction="row" spacing={3} alignItems="center">
+    <LegendLine color={legendColor} dash={dash} />
+    <Typography variant="bodySRegular">{label}</Typography>
+  </Stack>
+)
+
+const RevenueChartFooter = ({ timeOptions, activeTimeOption, setActiveTimeOption }: Props) => {
+  const {
+    design: { Color },
+  } = useTheme()
+  const averageLineColor = Color.Tertiary[400]
+  const sevenDayAverageLineColor = Color.Secondary[500]
+  const mainLineColor = Color.Primary[500]
```

Comment:
Using a map is more readable and typescript will error out if new keys are introduced

```suggestion
  export const priceLineColors = {
    proj_apy: Color.Primary[500],
    proj_apy_7d_avg: Color.Secondary[500],
    proj_apy_total_avg: Color.Tertiary[400],
  } as const satisfies Record<YieldKeys, string>
```